templates/load-message.lua

```diff
@@ -0,0 +1,57 @@
+-- Some parameters to control the behavior. Can be set via Redis keys.
+-- How long (in seconds) should a given buffer bucket be?
+local pBufferBucketInterval = tonumber(redis.call("GET", "conf-bufferBucketInterval")) or 60
+-- How many buffer buckets should we retain?
+local pBufferBucketCount = tonumber(redis.call("GET", "conf-bufferBucketCount")) or 20
+
+-- Derived parameters
+local pBufferLifetime = pBufferBucketInterval * pBufferBucketCount
+
+-- First, parse and restructure the incoming message.
+local message = cjson.decode(ARGV[1])
```

Comment:
We've seen partial messages sent from logstash-forwarder before. All I remember is that we needed to add [this line](https://github.com/aaw/logstash-output-syslog/blob/aptible/lib/logstash/outputs/syslog.rb#L124) to avoid logstash exceptions from incomplete prefixes of Docker log lines at one point. So it may be worth handling the failures here unless that's already expected by this script and the exception will just bubble up and cancel the processing by design.